File: my-folder/problems/search_a_2d_matrix_ii/solution.js
var searchMatrix = function(matrix, target) {
  let rowIndex = 0,
    columnIndex = matrix[0].length - 1;

  while (rowIndex < matrix.length && columnIndex >= 0) {
    if (matrix[rowIndex][columnIndex] === target) return true;
    if (matrix[rowIndex][columnIndex] < target) rowIndex++;
    else columnIndex--;
  }
  return false;
};

File: my-folder/problems/search_a_2d_matrix_ii/solution.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int rowIndex = 0, columnIndex = matrix[0].size() - 1;

        while (rowIndex < matrix.size() && columnIndex >= 0) {
            if (matrix[rowIndex][columnIndex] == target)
                return true;
            if (matrix[rowIndex][columnIndex] < target)
                rowIndex++;
            else
                columnIndex--;
        }
        return false;
    }
};